utils/install-open-shift-spire.sh

```diff
@@ -0,0 +1,222 @@
+#!/bin/bash
+# get the SCRIPT and TSI ROOT directories
+SCRIPT_PATH=$( cd "$(dirname "${BASH_SOURCE[0]}")" ; pwd -P )
+TSI_ROOT="${SCRIPT_PATH}/.."
+TSI_VERSION=$(cat ${SCRIPT_PATH}/../tsi-version.txt)
+
+# SPIRE Server info:
+SPIRESERVERPROJECT="tornjak"
+CLUSTERNAME=$1
+SPIRESERVER=$2
+# SPIRE Agent info:
+TRUSTD="${3:-spiretest.com}"
+PROJECT="${4:-spire}"
+SPIREGROUP="spiregroup"
+SPIREAGSA="spire-agent"
+SPIREAGSCC="spire-agent"
+
+## create help menu:
+helpme()
+{
+  cat <<HELPMEHELPME
+Install SPIRE agent and workload registrar for TSI
+
+Syntax: ${0} <CLUSTER_NAME> <SPIRE_SERVER> <PROJECT_NAME>
+
+Where:
+  CLUSTER_NAME - name of the OpenShift cluster (required)
+  SPIRE_SERVER - SPIRE server end-point (required)
+  TRUST_DOMAIN - the trust root of SPIFFE identity provider, default: spiretest.com (optional)
+  PROJECT_NAME - OpenShift project [namespace] to install the Agent, default: spire (optional)
+HELPMEHELPME
+}
+
+# validate the arguments
+if [[ "$1" == "-?" || "$1" == "-h" || "$1" == "--help" ]] ; then
+  helpme
+  exit 0
+elif [[ "$2" == "" ]] ; then
+  echo "Both CLUSTER_NAME and SPIRE_SERVER are required"
+  helpme
+  exit 1
+fi
+
+installSpireAgent(){
+  oc get projects | grep $PROJECT
+  if [ "$?" == "0" ]; then
+    # check if spire-agent project exists:
+    echo "$PROJECT project already exists. "
+    read -n 1 -r -p "Do you want to re-install it? [y/n]" REPLY
+    echo # create a new line
+    if [[ $REPLY =~ ^[Yy]$ || $REPLY == "" ]] ; then
+      echo "Re-installing $PROJECT project"
+      # oc delete project $PROJECT
+      cleanup
+      while (oc get projects | grep -v spire-server | grep $PROJECT); do echo "Waiting for $PROJECT removal to complete"; sleep 2; done
```

Comment:
i think we can end the block here and just make it always run the two lines below by removing the `if [ "$?" == "0" ]` else block, to remove code duplication

```
 oc new-project $PROJECT --description="My TSI Spire Agent project on OpenShift" > /dev/null
oc project $PROJECT```

Reply:
I am not sure I follow. There are two initial options: either the project exists or not, and if exists we should check if they want to reinstall. I think your suggestion would ask for the reinstall without checking if it exists